src/com/tw/go/plugin/model/GitConfig.java

```diff
@@ -42,10 +42,25 @@ public String getEffectiveUrl() {
         }
         return getUrl();
     }
+    
+    public String getEffectiveMaskedUrl() {
+        if (isRemoteUrl() && hasCredentials()) {
+            return getUrlWithMaskedCredentials();
+        }
+        return getUrl();
+    }
+    
+    private String getUrlWithCredentials(String user, String pass) {
+        String[] parts = url.split("://");
+        return String.format("%s://%s:%s@%s", parts[0], user, pass, parts[1]);
+    }
 
     public String getUrlWithCredentials() {
-        String[] parts = url.split("://");
-        return String.format("%s://%s:%s@%s", parts[0], username, password, parts[1]);
+        return getUrlWithCredentials(username, password);
+    }
+    
+    public String getUrlWithMaskedCredentials() {
+        return getUrlWithCredentials(username, StringUtil.repeat("*", password.length()));
```

Comment:
adding number of `*` = `password.length()` could be a problem. its giving away length of password. You might want to keep that fixed to say 5 `*`?